api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -1241,6 +1249,33 @@ public void testNotebookFileList() throws Exception {
     assertEquals(gotNames, ImmutableList.of("mockFile.ipynb", "two words.ipynb"));
   }
 
+  @Test
+  public void testAsyncCallFromNotebookFileList() throws Exception {
+    when(fireCloudService.getWorkspace("project", "workspace")).thenReturn(
+        new org.pmiops.workbench.firecloud.model.WorkspaceResponse()
+            .workspace(
+                new org.pmiops.workbench.firecloud.model.Workspace()
+                    .bucketName("bucket")));
+    doNothing()
+        .when(userRecentResourceService)
+        .deleteOrphanNotebookEntries(anyLong(), anyLong(), anyList());
+    Blob mockBlob1 = mock(Blob.class);
+    Blob mockBlob2 = mock(Blob.class);
+    when(mockBlob1.getName()).thenReturn("notebooks/mockFile.ipynb");
+    when(mockBlob2.getName()).thenReturn("notebooks/two words.ipynb");
+    when(cloudStorageService.getBlobList("bucket", "notebooks")).thenReturn(
+        ImmutableList.of(mockBlob1, mockBlob2));
+    workspacesController
+        .getNoteBookList("project", "workspace").getBody();
+
+    CompletableFuture<Void> asyncHandleOrphanNotebook = workspacesController.getAsyncHandleOrphanNotebooks();
+    Thread.sleep(5000);
+    assertEquals(asyncHandleOrphanNotebook.isDone(), true);
+    asyncHandleOrphanNotebook.thenAccept(result -> {
```

Comment:
This is interesting, I'm not sure I follow why it's necessary - it seems like either the below is executing synchronously (and therefore the verification runs) - implying you don't need the callback OR the callback runs asynchronously in which case the verification wouldn't run within this test case.